File: 4A/ArithmeticMeanVBNET/ArithmeticMeanVBNET/RandomPerson.vb
Imports System
Imports System.CodeDom.Compiler
Imports System.Collections.Generic
Imports System.Text
Imports System.Net.Http
Imports System.Threading.Tasks
Imports System.IO
Imports System.Net
Imports System.Web.Script.Serialization

Public Class Education
    Public Property certificate As String
    Public Property university As String
End Class

Public Class Marriage
    Public Property married As Boolean
End Class

Public Class OnlineInfo
    Public Property email As String
    Public Property ip_address As String
    Public Property ipv6_address As String
    Public Property password As String
    Public Property password_md5 As String
    Public Property user_agent As String
    Public Property username As String
End Class

Public Class Personal
    Public Property age As Integer
    Public Property blood As String
    Public Property born As Object
    Public Property born_place As String
    Public Property cellphone As String
    Public Property city As String
    Public Property country As String
    Public Property eye_color As String
    Public Property father_name As String
    Public Property gender As String
    Public Property height As String
    Public Property last_name As String
    Public Property name As String
    Public Property national_code As String
    Public Property religion As String
    Public Property system_id As String
    Public Property weight As Integer
End Class

Public Class Work
    Public Property country_code As String
    Public Property insurance As Boolean
    Public Property position As String
    Public Property salary As String
End Class

Public Class Person
    Public Property education As Education
    Public Property marriage As Marriage
    Public Property online_info As OnlineInfo
    Public Property personal As Personal
    Public Property work As Work
End Class

Public Class ExamplePerson
    Public Property person As Person
End Class

Public Class RandomPerson
    Public Shared url As String = "https://pipl.ir/v1/getPerson"

    Public Shared Function Generate() As Person
        Dim jsonString As String = String.Empty
        Dim request As HttpWebRequest = CType(WebRequest.Create(url), HttpWebRequest)
        request.AutomaticDecompression = DecompressionMethods.GZip

        Using response As HttpWebResponse = CType(request.GetResponse(), HttpWebResponse)

            Using stream As Stream = response.GetResponseStream()

                Using reader As StreamReader = New StreamReader(stream)
                    jsonString = reader.ReadToEnd()
                End Using
            End Using
        End Using

        Dim jss As New JavaScriptSerializer()
        Dim EP As ExamplePerson = jss.Deserialize(Of ExamplePerson)(jsonString)
        Return EP.person
    End Function
End Class

File: 4A/ArithmeticMeanCSharp/ArithmeticMeanCSharp/RandomPerson.cs
using System;
using System.CodeDom.Compiler;
using System.Collections.Generic;
using System.Text;
using System.Net.Http;
using System.Threading.Tasks;
using System.Text.Json;
using System.Text.Json.Serialization;
using System.IO;
using System.Net;

namespace ArithmeticMeanCSharp
{

    public class Education
    {
        public string certificate { get; set; }
        public string university { get; set; }
    }

    public class Marriage
    {
        public bool married { get; set; }
    }

    public class OnlineInfo
    {
        public string email { get; set; }
        public string ip_address { get; set; }
        public string ipv6_address { get; set; }
        public string password { get; set; }
        public string password_md5 { get; set; }
        public string user_agent { get; set; }
        public string username { get; set; }
    }

    public class Personal
    {
        public int age { get; set; }
        public string blood { get; set; }
        public object born { get; set; }
        public string born_place { get; set; }
        public string cellphone { get; set; }
        public string city { get; set; }
        public string country { get; set; }
        public string eye_color { get; set; }
        public string father_name { get; set; }
        public string gender { get; set; }
        public string height { get; set; }
        public string last_name { get; set; }
        public string name { get; set; }
        public string national_code { get; set; }
        public string religion { get; set; }
        public string system_id { get; set; }
        public int weight { get; set; }
    }

    public class Work
    {
        public string country_code { get; set; }
        public bool insurance { get; set; }
        public string position { get; set; }
        public string salary { get; set; }
    }

    public class Person
    {
        public Education education { get; set; }
        public Marriage marriage { get; set; }
        public OnlineInfo online_info { get; set; }
        public Personal personal { get; set; }
        public Work work { get; set; }
    }

    public class ExamplePerson
    {
        public Person person { get; set; }
    }

    public class RandomPerson
    {
        
        static public string url = "https://pipl.ir/v1/getPerson";

        static public Person Generate()
        {

            string jsonString = string.Empty;

            
            HttpWebRequest request = (HttpWebRequest)WebRequest.Create(url);
            request.AutomaticDecompression = DecompressionMethods.GZip;

            using (HttpWebResponse response = (HttpWebResponse)request.GetResponse())
            using (Stream stream = response.GetResponseStream())
            using (StreamReader reader = new StreamReader(stream))
            {
                jsonString = reader.ReadToEnd();
            }

            ExamplePerson EP = JsonSerializer.Deserialize<ExamplePerson>(jsonString);
            return EP.person;

        }
    }

}
